app/Jobs/CreateQueryserviceBatchesJob.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Jobs;
+
+use App\EventPageUpdate;
+use App\QsBatch;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Config;
+
+class CreateQueryserviceBatchesJob extends Job
+{
+    private const NAMESPACE_ITEM = 120;
+    private const NAMESPACE_PROPERTY = 122;
+    private const NAMESPACE_LEXEME = 146;
+
+    private int $entityLimit;
+
+    public function __construct()
+    {
+        $this->entityLimit = Config::get('wbstack.qs_batch_entity_limit');
+    }
+
+    public function handle(): void
+    {
+        DB::transaction(function () {
+            // Get ID of the latest batch that we have created (or 0)
+            $maxIdBatch = QsBatch::orderBy('id', 'desc')->first();
```

Comment:
I found this a bit confusing when I was debugging last night; of course it's actually fine but I was wondering a bit how this works in cases where there are no undone batches. If you had a smart idea how to make this better I'd be all ears (add "consumed" flag to the event? or store this number in it's own place?). If not no pressure; I can't easily thing of a better way to do this

Reply:
I share the sentiment, but I also haven't got a very good idea about how to change this right now. We'd need some sort of "cursor", but I have zero clue where to store this.